samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -65,11 +65,21 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
+  private int currentGeneration;
 
   public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
     this.keyBuilder = zkKeyBuilder;
     this.connectionTimeoutMs = connectionTimeoutMs;
     this.zkClient = zkClient;
+    currentGeneration = 0;
```

Comment:
It might be worth a note in the class docs about how currentGeneration is used in ZkUtil.